### PR TITLE
(maint) Add minimum ruby version and specification to gemspec

### PR DESCRIPTION
**Pull request by haus**
Because of the use of keyword arguments with defaults, vanagon won't
work on anything less than ruby 2.1.0, so this commit updates the
gemspec to reflect that. We also add the specification version to
enforce a relatively modern version of rubygems.

### CODE REVIEW

File: vanagon.gemspec [hunk 1]
```diff
@@ -12,6 +12,8 @@ Gem::Specification.new do |gem|
   gem.authors  = ['Puppet Labs']
   gem.email    = 'info@puppetlabs.com'
   gem.homepage = 'http://github.com/puppetlabs/vanagon'
+  gem.specification_version = 3
+  gem.required_ruby_version = '~> 2.1.0'
```
Comment (underscorgan) [inline vanagon.gemspec]: Do we want this this tightly bound or do we expect it to work with any 2.x >= 2.1.0?

Comment (haus) [inline vanagon.gemspec]: @underscorgan I don't expect it to break on a 2.x release. it works fine on 2.2, but I'm open to all of the possibilities.

Comment (underscorgan) [inline vanagon.gemspec]: @haus ok, sounds reasonable.

**Comment by underscorgan** [comment]
:+1:

**Pull request merged**